qrcode_detection_package/main.py

```
@@ -1,56 +1,358 @@
+        are logged. Additionally, the relative midpoint coordinates, relative to the center of the image, are calculated
+        and logged. If no QR code is detected, the method returns an error message along with default midpoint coordinates.
+
+        Args: 
+            image: The image in which to detect QR codes in the OpenCV MatLike format.
+
+        Returns: 
+            A tuple containing:
+                - The decoded information from the QR code, or "Error" if no QR code is detected.
+                - The x-coordinate of the geometric midpoint of the detected QR code.
+                - The y-coordinate of the geometric midpoint of the detected QR code.
+        """
+        # detect and decode QR codes in the image using OpenCV library
         decoded_info, points, _ = self.qr_code_detector.detectAndDecode(image)
 
-        return decoded_info
+        # check if QR code was successfully decoded
+        if (len(decoded_info) > 0):
+            # Log QR-Code content
+            self.get_logger().info(f"Detected QR code: {decoded_info}")
+
+            # Calculate the geometric midpoint of the bounding rectangle
+            midpoint_x, midpoint_y = self.__corners_to_middlepoint(points)
+
+            # Log the midpoint coordinates
+            self.get_logger().info(
+                f"QR code middle point: ({midpoint_x}|{midpoint_y})")
+
+            # Get midpoints relative to the center of the picture
+            img_height, img_width = image.shape[:2]
+            rel_midpoint_x, rel_midpoint_y = self.__relative_midpoint(
+                midpoint_x, midpoint_y, img_width, img_height)
+
+            # Log the relative midpoint coordinates
+            self.get_logger().info(
+                f"Relative QR code middle point: ({rel_midpoint_x}|{rel_midpoint_y})")
+        else:
+            # if no QR-Code was detected set return values to error standard
+            decoded_info = "Error"
```

You could raise an Exception instead of returning default values. But that's up to you